diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/NewlinesRule.kt

```diff
@@ -313,6 +337,12 @@ class NewlinesRule(private val configRules: List<RulesConfig>) : Rule("newlines"
     private fun ASTNode.getParentExpressions() =
             parents().takeWhile { it.elementType in chainExpressionTypes && it.elementType != LAMBDA_ARGUMENT }
 
+    private fun isMultilineLambda(node: ASTNode): Boolean {
+        val leftBrace = node.findAllNodesWithSpecificType(LBRACE).firstOrNull()
+
+        return leftBrace != null && leftBrace.treeNext.isWhiteSpaceWithNewline()
```

Comment:
There can be a lambda with parameters after `{`. You could, maybe, just check node type of lambda and then count lines of text.